src/views/1_ApplicationComponent.tsx

```diff
@@ -140,6 +141,20 @@ export default class ApplicationComponent extends React.Component<{}, State> {
         );
     }
 
+    closeSelectedTab(event: any) {
+        let selectedTabIndex: number = Number(event.target.value);
+
+        if (selectedTabIndex === this.activeTabIndex) {
+            this.closeSession(this.activeTab.activeSession());
+        } else {
+            this.closeTab(this.tabs[selectedTabIndex]);
+            if (selectedTabIndex < this.activeTabIndex)
+                this.activeTabIndex -= 1;
+            this.setState({sessions: this.activeTab.sessions});
+        }
```

Comment:
It really should be the same as `event.metaKey && event.keyCode === KeyCode.W` handling (except selecting the  correct tab to close).